tests/Unit/EmogrifierTest.php

```diff
@@ -737,6 +737,117 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
         static::assertContains(sprintf($expectedHtml, $cssDeclaration1, $cssDeclaration2), $result);
     }
 
+    /**
+     * @return string[][]
+     */
+    public function differentCssSelectorSpecificityDataProvider()
+    {
+        return [
+            'class more specific than types' => ['<p class="p-1"', 'html body p', '.p-1'],
+            'ID more specific than class and types' => ['<p class="p-4" id="p4"', 'html body .p-4', '#p4'],
```

Comment:
Do we need two types here, or would one suffice? (Test cases should be minimal so only relevant things are there, making it easier to find out what's important when reading the test.)